app/src/main/java/com/example/recipeopedia/activities/ViewAttemptsActivity.java

```diff
@@ -0,0 +1,123 @@
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.recipeopedia.R;
+import com.example.recipeopedia.adapters.AttemptAdapter;
+import com.example.recipeopedia.models.Attempt;
+import com.example.recipeopedia.models.FavoriteRecipe;
+import com.example.recipeopedia.models.Review;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import org.parceler.Parcels;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViewAttemptsActivity extends AppCompatActivity {
+    private Button btnPostAttempt;
+    private EditText etYourAttempt;
+    private RecyclerView rvAttempts;
+    private FavoriteRecipe favoriteRecipe;
+    private SwipeRefreshLayout swipeContainer;
+    protected AttemptAdapter attemptAdapter;
+    protected List<Attempt> mAttempts;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_attempts);
+
+        swipeContainer = findViewById(R.id.swipeContainer);
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                attemptAdapter.clear();
+                queryAttempts();
+                swipeContainer.setRefreshing(false);
+            }
+        });
+
+        mAttempts = new ArrayList<>();
+        attemptAdapter = new AttemptAdapter(mAttempts);
+        rvAttempts = findViewById(R.id.rvAttempts);
+        rvAttempts.setAdapter(attemptAdapter);
+        rvAttempts.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+        queryAttempts();
+
+        favoriteRecipe = Parcels.unwrap(getIntent().getParcelableExtra(FavoriteRecipe.class.getSimpleName()));
+        etYourAttempt = findViewById(R.id.etYourAttempt);
+        btnPostAttempt = findViewById(R.id.btnPostAttempt);
+        ParseUser currentUser = ParseUser.getCurrentUser();
+        btnPostAttempt.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                favoriteRecipe = Parcels.unwrap(getIntent().getParcelableExtra(FavoriteRecipe.class.getSimpleName()));
+                String description = etYourAttempt.getText().toString();
+                if (description.isEmpty()) {
+                    Toast.makeText(getApplicationContext(), "Description cannot be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                else {
+                    try {
+                        saveAttempt(favoriteRecipe.getRecipeName(), currentUser, description);
+                        attemptAdapter.clear();
+                        queryAttempts();
```

Comment:
just FYI, for a production app i would set this "optimistically" rather than clear and requery, but this is fine for an intern project